src/vm/contexts.rs

```diff
@@ -63,6 +83,246 @@ pub struct CallStack {
 
 pub type StackTrace = Vec<FunctionIdentifier>;
 
+impl AssetMap {
+    pub fn new() -> AssetMap {
+        AssetMap {
+            token_map: HashMap::new(),
+            asset_map: HashMap::new()
+        }
+    }
+
+    // This will get the next amount for a (principal, asset) entry in the asset table.
+    fn get_next_amount(&self, principal: &PrincipalData, asset: &AssetIdentifier, amount: i128) -> Result<i128> {
+        let current_amount = match self.token_map.get(principal) {
+            Some(principal_map) => *principal_map.get(&asset).unwrap_or(&0),
+            None => 0
```

Comment:
Is it ever okay for this to be `None`?

Reply:
Yep, this happens the first time a token transfer is logged for a given principal, since the principal doesn't yet appear in the HashMap, it's starting balance is 0.